packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-alternative-methods-button-strategy.ts

```diff
@@ -0,0 +1,143 @@
+        const currency = state.cart.getCartOrThrow().currency.code;
+        this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        this._paypalCommerceSdk = await this._paypalScriptLoader.loadPaypalCommerce(this._paymentMethod, currency, initializesOnCheckoutPage);
+
+        this._renderButton(apm, containerId, initializesOnCheckoutPage, style);
+    }
+
+    deinitialize(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    private _renderButton(
+        apm: string,
+        containerId: string,
+        initializesOnCheckoutPage?: boolean,
+        style?: PaypalButtonStyleOptions
+    ): void {
+        const paypalCommerceSdk = this._getPayPalCommerceSdkOrThrow();
+        const isAvailableFundingSource = Object.values(paypalCommerceSdk.FUNDING).includes(apm);
+
+        if (!isAvailableFundingSource) {
+            throw new InvalidArgumentError(`Unable to initialize PayPal button because "options.paypalcommercealternativemethods.apm" argument is not valid funding source.`);
+        }
+
+        const validButtonStyle = style ? this._getButtonStyle(style) : {};
+
+        const buttonRenderOptions: ButtonsOptions = {
+            fundingSource: apm,
+            style: validButtonStyle,
+            createOrder: () => this._createOrder(initializesOnCheckoutPage),
+            onApprove: ({ orderID }: ApproveDataOptions) => this._tokenizePayment(orderID),
+        };
+
+        const paypalButtonRender = paypalCommerceSdk.Buttons(buttonRenderOptions);
+
+        if (paypalButtonRender.isEligible()) {
+            paypalButtonRender.render(`#${containerId}`);
+        } else {
+            this._removeElement(containerId);
+        }
+    }
+
+    private async _createOrder(initializesOnCheckoutPage?: boolean): Promise<string> {
+        const state = this._store.getState();
+        const cart = state.cart.getCartOrThrow();
+
+        const providerId = initializesOnCheckoutPage ? 'paypalcommercealternativemethodscheckout' : 'paypalcommercealternativemethod';
+
+        const { orderId } = await this._paypalCommerceRequestSender.createOrder(cart.id, providerId);
+
+        return orderId;
+    }
+
+    private _tokenizePayment(orderId?: string): void {
+        const paymentMethod = this._getPaymentMethodOrThrow();
+
+        if (!orderId) {
+            throw new MissingDataError(MissingDataErrorType.MissingOrderId);
+        }
+
+        return this._formPoster.postForm('/checkout.php', {
+            payment_type: 'paypal',
+            action: 'set_external_checkout',
+            provider: paymentMethod.id,
+            order_id: orderId,
+        });
+    }
+
+    private _getPayPalCommerceSdkOrThrow(): PaypalCommerceSDK {
+        if (!this._paypalCommerceSdk) {
+            throw new PaymentMethodClientUnavailableError();
+        }
+
+        return this._paypalCommerceSdk;
+    }
+
+    private _getPaymentMethodOrThrow(): PaymentMethod {
```

Comment:
I am using this method to check if there any value in `this_paymentMethod` global class variable. It can help us to avoid passing extra `methodId` prop through all functions to get this variable where we need it.

Reply:
And I have used the same method in all updated PayPal Commerce checkout button strategies as well

Reply:
Just out of curiosity why do we need to add the global class variable? We should be ok to resolve the paymentMethod in initialize and pass it to relevant functions.

Reply:
Yeah. Ok. I will update this one